src/base-command.ts

```diff
@@ -115,6 +115,7 @@ export default abstract class BaseCommand extends Command {
 
   async catch(error: any): Promise<void> {
     if (error.oclif && error.oclif.exit === 0) return;
+    process.exitCode = 1;
```

Comment:
I still find setting the exitCode a little suspect. 

Is this because we are returning for the prettyValidationErrors?

```ts
if (error instanceof ValidationErrors) {
  return prettyValidationErrors(error);
}
```

Reply:
I think that is correct. we are skipping the super.catch call, (https://github.com/oclif/core/blob/main/src/command.ts#L253) and the oclif cleanup doesn't appear to happen during finally; terminating with exit code 0.
Other errors seem to exit with error codes fine. Perhaps it's because super.catch is otherwise always called. I can look into this further into suppressing the output from super.catch when called and/or move this into the conditional as an edge case;

Reply:
If its only for the pretty errors then the solution should be more targeted.

```ts
if (error instanceof ValidationErrors) {
  prettyValidationErrors(error);
  process.exit(1);
}
```

or potentially calling super.catch, but then the resulting error wouldn't be super pretty :P

Reply:
By adding the optional property exitCode to the validationErrors object, we can return super.catch({...error}), to get the best of both worlds.  pretty validation errors, and returning the super.catch() fn (generating the expected error exit code)

updated PR.